### DISCUSSION TITLE
External Libraries in MOOSE

### DISCUSSION THREAD

**Discussion by AmerMalik9** [idaholab/moose#18096] [Q&A General] [answered]
Hi,
We are planning to use MOOSE for our commercial software and for that we need to make sure that we don't violate any of the license rules both for MOOSE and for all the external libraries within MOOSE. Could anyone provide us with the complete list of external libraries that are being used within MOOSE. I found this list: https://mooseframework.inl.gov/sqa/library_requirements.html but I am not sure if it's complete or updated?
Thank you!

**Comment by milljm** [comment, marked as answer]
The items on the provided list look up to date and accurate to my eyes.

**Reply by doquang** [reply]
@AmerMalik9 - MOOSE uses PETSc, and the default configuration of PETSc on MOOSE requires:
+ Hyper
+ fblaslapack
+ metis
+ ptscotch
+ parmetis
+ superlu
+ mumps
+ strumpack
+ scalapack
+ slepc
(See moose/scripts/update_and_rebuild_petsc.sh)

**Reply by AmerMalik9** [reply]
@doquang Thanks Minh! Will get back to you for more questions.
@milljm Thanks for the confirmation.

**Reply by aeslaughter** [reply]
You can also take a look at our manual install instructions to control/follow everything you need: https://mooseframework.inl.gov/getting_started/installation/manual_installation_gcc.html

Our default builds for libMesh and PETSc can be found in our scripts directory, these scripts should be relying on anything compatible. However, it is possible to configure them with things that are not.
https://github.com/idaholab/moose/blob/next/scripts/update_and_rebuild_libmesh.sh
https://github.com/idaholab/moose/blob/next/scripts/update_and_rebuild_petsc.sh

Finally, framework/contrib contains some software packages that we rely on as well. To the best of our knowledge everything is compatible with LGPL 2.1.
